plugins/s3_auth/aws_auth_v4.cc

```diff
@@ -256,7 +285,15 @@ getCanonicalRequestSha256Hash(TsInterface &api, bool signPayload, const StringSe
     String encodedParam = uriEncode(param, /* isObjectName */ false);
 
     paramNames.insert(encodedParam);
-    paramsMap[encodedParam] = uriEncode(value, /* isObjectName */ false);
+    String decodedValue = uriDecode(value);
+    if (0 == decodedValue.compare(value)) {
```

Comment:
Is there no way to check if the string is already encoded, other than doing the full decode and then do the string compare ? Look for % ?

Reply:
The lack of `%` would mean that the string is not URI-encoded but the presence of `%` would not guarantee that the string is uri-encoded, decoding and comparing feels like a much better proof.

I guess I could optimize by adding a check for `%` to try to avoid uri-decoding of strings not containing `%`.

But I am not sure how often we would have uri-encoded query parameter values in the general case (most of the requests in our use case are url-encoded) and I am not sure if I can reliably judge if this is something worth optimizing (it seems in our case we would end up with an extra search for `%` for most of the requests)